tests/console/commands/plugin/test_add.py

```diff
@@ -161,6 +161,38 @@ def test_add_with_git_constraint_with_extras(
     )
 
 
+def test_add_with_git_constraint_with_subdirectory(
+    app: PoetryTestApplication,
+    repo: TestRepository,
+    tester: CommandTester,
+    env: MockEnv,
+    installed: Repository,
+):
+    repo.add_package(Package("pendulum", "2.0.5"))
+
+    tester.execute("git+https://github.com/demo/poetry-plugin2.git#subdirectory=subdir")
```

Comment:
Can we add parameterised test cases that evaluate this for cases where a revision or branch is also specified?

Reply:
@abn I've added the required tests. As I mentioned above, git urls with subdirectory and rev will fail due to the regex issue in poetry core.